erpnext/hr/doctype/employee_grievance/employee_grievance.py

```diff
@@ -0,0 +1,112 @@
+# Copyright (c) 2021, Frappe Technologies Pvt. Ltd. and contributors
+# For license information, please see license.txt
+
+import frappe
+from frappe import _, bold
+from frappe.utils import comma_and, add_days, get_datetime, today
+from frappe.model.document import Document
+
+class EmployeeGrievance(Document):
+	def validate(self):
+		self.check_cause_of_grievance()
+		self.check_resolution_details()
+
+	def on_submit(self):
+		if self.status not in ["Invalid", "Resolved"]:
+			frappe.throw(_("Only Employee Grievance with status {0} and {1} can be submitted").format(
+				bold("Invalid"),
+				bold("Resolved"))
+			)
+
+	def before_cancel(self):
+		if self.suspended_from and self.suspended_to:
+			frappe.throw(_("You need to Unsuspend Employee before Cancellation"))
+
+	def check_cause_of_grievance(self):
+		if self.status == "Investigated" and not self.cause_of_grievance:
+			frappe.throw(_("You ned to set Cause of Grievance before Setting Status to Investigated"))
+
+	def check_resolution_details(self):
+		raise_error = 0
+		mandatory_fields = []
+		if self.status == "Resolved":
+			if not self.resolution_date:
+				mandatory_fields.append("Resolution Date")
+				raise_error = 1
+
+			if not self.resolved_by:
+				mandatory_fields.append("Resolved By")
+				raise_error = 1
+
+			if not self.resolution_detail:
+				mandatory_fields.append("Resolution Detail")
+				raise_error = 1
+
+			if not self.cause_of_grievance:
+				mandatory_fields.append("Cause of Grievance")
+				raise_error = 1
+
+
+		fields_text = bold(comma_and(mandatory_fields))
+		if raise_error:
+			frappe.throw(_("You need to set {0} before setting Employee Grievance Resolved").format(fields_text))
```

Comment:
@ruchamahabal


- "**Description**" is actually what happened to the employee.  
- "**Cause of grievance**" is like, why this grievance happened in the first place, finding the root cause.  it is the result of the investigation
-  "**resolution detail**" are details for solution and prevention for the problem raised by the employee. so that, this grievance will not be happening again to employees.
-  "**action**' is a punishment given like Suspension and Pay Cut

I think these are the important matrices to be captured for complete grievance resolution.
for example, I filled the grievance form please refer to the screenshot in the description.

Reply:
Okay, then we should add the label "Investigation Details" for the Cause of Grievance section. So there will be a clear distinction:

1. Description
2. Investigation
3. Action
4. Resolution